File: WinForms/VB/PivotGridMainDemo/Modules/Filtering.vb
Imports System
Imports System.ComponentModel
Imports System.Drawing
Imports DevExpress.Data.Filtering
Imports System.Globalization

Namespace DevExpress.XtraPivotGrid.Demos.Modules

    Public Partial Class Filtering
        Inherits TutorialControl

        Private initialized As Boolean

        Public Sub New()
            CreateWaitDialog()
            InitializeComponent()
            pivotGridControl1.DataSource = GetNWindData("SalesPerson")
            initialized = True
            ApplyFilter()
        End Sub

        Public Overrides ReadOnly Property ViewOptionsControl As PivotGridControl
            Get
                Return pivotGridControl1
            End Get
        End Property

        Public Overrides ReadOnly Property ExportControl As PivotGridControl
            Get
                Return pivotGridControl1
            End Get
        End Property

        '<dateEdit1>
        '<dateEdit2>
        Private Sub dateEdit_EditValueChanged(ByVal sender As Object, ByVal e As EventArgs)
            If initialized Then ApplyFilter()
        End Sub

        Private Sub ApplyFilter()
            Dim str1 As String = GetCriteria(dateEdit1.DateTime, True), str2 As String = GetCriteria(dateEdit2.DateTime, False)
            If Not String.IsNullOrEmpty(str1) AndAlso Not String.IsNullOrEmpty(str2) Then
                pivotGridControl1.ActiveFilterCriteria = CriteriaOperator.Parse(str1 & " And " & str2)
            Else
                pivotGridControl1.ActiveFilterCriteria = CriteriaOperator.Parse(str1 & str2)
            End If
        End Sub

        Private Function GetCriteria(ByVal [date] As Date, ByVal isGreater As Boolean) As String
            If [date].Ticks = 0 Then Return ""
            Return String.Format("{0} {1} #{2}#", fieldOrderDate.FilterColumnName, If(isGreater, ">=", "<="), Convert.ToString([date], CultureInfo.InvariantCulture))
        End Function
    '</dateEdit2>
    '</dateEdit1>
    End Class
End Namespace

File: WinForms/CS/PivotGridMainDemo/Modules/Filtering.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Drawing;
using System.Data;
using System.Text;
using System.Windows.Forms;
using DevExpress.Data.Filtering;
using System.Globalization;

namespace DevExpress.XtraPivotGrid.Demos.Modules {
	public partial class Filtering : TutorialControl {
        bool initialized;

		public Filtering() {
			CreateWaitDialog();
			InitializeComponent();

			pivotGridControl1.DataSource = GetNWindData("SalesPerson");
            initialized = true;
            ApplyFilter();
		}
		public override PivotGridControl ViewOptionsControl { get { return pivotGridControl1; } }
		public override PivotGridControl ExportControl { get { return pivotGridControl1; } }

        //<dateEdit1>
        //<dateEdit2>
		private void dateEdit_EditValueChanged(object sender, EventArgs e) {
            if(initialized)
			    ApplyFilter();
		}

		void ApplyFilter() {
			string str1 = GetCriteria(dateEdit1.DateTime, true),
				str2 = GetCriteria(dateEdit2.DateTime, false);

			if(!string.IsNullOrEmpty(str1) && !string.IsNullOrEmpty(str2))
				pivotGridControl1.ActiveFilterCriteria = CriteriaOperator.Parse(str1 + " And " + str2);
			else
				pivotGridControl1.ActiveFilterCriteria = CriteriaOperator.Parse(str1 + str2);
		}        
		string GetCriteria(DateTime date, bool isGreater) {
			if(date.Ticks == 0) return "";
			return string.Format("{0} {1} #{2}#", fieldOrderDate.FilterColumnName, isGreater ? ">=" : "<=", 
                Convert.ToString(date, CultureInfo.InvariantCulture));
		}
        //</dateEdit2>
        //</dateEdit1>
	}
}
